### PR TITLE
DCS-1895 Add mandatory check warning page in DM decision process

### PR DESCRIPTION
**Pull request by davidthomas22**

### CODE REVIEW

File: assets/sass/application.scss [hunk 1]
```diff
@@ -450,6 +450,12 @@ input[type="time"] {
   background-color: #FFF0D5;
 }
 
+.mandatory-warning-container {
+  @extend %container-shared;
+  background-color: #ffffff;
+  padding-left: 0%;
+}
+
```
Comment (davidthomas22) [inline assets/sass/application.scss:458]: Reluctant to keep in the `application.scss` file but decided to keep here to maintain consistency as the other container classes are maintained here

File: licences-specs/src/test/resources/licences/decision/mandatory-check-required.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          "approved": "Yes"
+        }
+      ]
+    },
+    "risk": {
+      "riskManagement": {
+        "planningActions": "No",
+        "awaitingInformation": "No",
+        "proposedAddressSuitable": "Yes",
+        "showMandatoryAddressChecksNotCompletedWarning": "True"
```
Comment (davidthomas22) [inline licences-specs/src/test/resources/licences/decision/mandatory-check-required.json:97]: Added `mandatory-check-required.json`file and set `"showMandatoryAddressChecksNotCompletedWarning": "True"` to enable testing of the mandatory check page journey for the integration tests

File: server/routes/viewModels/taskLists/dmTasks.js [hunk 3]
```diff
@@ -58,5 +82,9 @@ module.exports = (licenceStatus) => {
     return rejectedAddressTaskList(licenceStatus)
   }
 
+  if (showMandatoryAddressChecksNotCompletedWarning) {
+    return mandatoryCheckTaskList(licenceStatus)
+  }
```
Comment (davidthomas22) [inline server/routes/viewModels/taskLists/dmTasks.js:87]: Added new mandatory check task to Decision maker flow and conditionally set to return mandatory check warning page route if `showMandatoryAddressChecksNotCompletedWarning` is `true`

File: test/routes/approval.test.js [hunk 4]
```diff
@@ -57,6 +57,7 @@ describe('/hdc/approval', () => {
     const routes = [
```
Comment (davidthomas22) [inline test/routes/approval.test.js:57]: Test to check new mandatory check page route and content

File: server/views/approval/mandatoryCheck.pug [hunk 5]
```diff
@@ -0,0 +1,42 @@
+extends ../layout
```
Comment (davidthomas22) [inline server/views/approval/mandatoryCheck.pug:1]: New mandatory check page

File: licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/decision/MandatoryCheckSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  def setupSpec() {
+    testData.loadLicence('decision/unstarted')
+    actions.logIn('DM')
+  }
+
+  def cleanupSpec() {
+    actions.logOut()
+  }
+
+  def 'Mandatory Check warning shown when mandatory risk check had not been completed '() {
```
Comment (davidthomas22) [inline licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/decision/MandatoryCheckSpec.groovy:30]: Work in progress to get integration tests to pass

File: licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/decision/MandatoryCheckSpec.groovy [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  def setupSpec() {
+    testData.loadLicence('decision/unstarted')
+    actions.logIn('DM')
+  }
+
+  def cleanupSpec() {
+    actions.logOut()
+  }
+
+  def 'Incomplete mandatory checks - continue journey'() {
```
Comment (davidthomas22) [inline licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/decision/MandatoryCheckSpec.groovy:29]: Integration tests to check the mandatory check page journeys. Tests Pass locally

Comment (louisenorris) [reply licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/decision/MandatoryCheckSpec.groovy:29]: Should there be a test for not showing the check page if mandatory checks have been completed?

File: licences-specs/src/test/resources/licences/decision/mandatory-check-required.json (hunk 2 above)
Comment (louisenorris) [reply licences-specs/src/test/resources/licences/decision/mandatory-check-required.json:97]: Might need to add `"version": "1"` to the `riskManagement` object as the version should always be present now once the migration is done.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**